apps/web-client/src/lib/db/cr-sqlite/history.ts

```diff
@@ -1,5 +1,38 @@
+/**
+ * @fileoverview Transaction History System
+ *
+ * History Overview:
+ * - Provides historical views of committed notes and transactions
+ * - Supports querying past notes by date
+ * - Supports querying past transactions with flexible filters
+ * - Calculates totals and pricing for historical analysis
+ * - Only includes committed notes/transactions
+ *
+ * Data Sources:
+ * - note table: Core note data and commit timestamps
+ * - book_transaction table: Individual book movements
+ * - book table: Book metadata and pricing
+ * - warehouse table: Location names and discounts
+ */
+
 import type { DB, PastNoteItem, PastTransactionItem } from "./types";
 
+/**
+ * Retrieves all committed notes for a specific date.
+ * Includes summary information like total books and pricing.
+ * Groups transactions by note and calculates warehouse-specific discounts.
+ *
+ * @param {DB} db - Database connection
+ * @param {string} date - Date to query in YYYY-MM-DD format
+ * @returns {Promise<PastNoteItem[]>} Committed notes with:
+ *   - id: Note identifier
+ *   - displayName: Human readable name
+ *   - noteType: "inbound" or "outbound"
+ *   - totalBooks: Sum of all book quantities
+ *   - warehouseName: Associated warehouse
+ *   - totalCoverPrice: Sum of (quantity * book price)
+ *   - totalDiscountedPrice: Sum of (quantity * book price * (1 - discount))
```

Comment:
I think we don't need to specify what's "inside" the types we're using.

```suggestion
 * @returns {Promise<PastNoteItem[]>} Committed notes
```